### PR TITLE
[Enhancement] Analyze job support exclude properties

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

Add statistics job properties: statistic_exclude_pattern, a regex, to forbidden some table to collect in a job. Example, don't collect `td.a1` table in analyze job

```
MySQL td> create analyze all properties ('statistic_exclude_pattern' ='td.a1');
Query OK, 0 rows affected
Time: 0.015s
MySQL td> show analyze;
+---------+----------+-------+---------+--------+----------+-----------------------------------+---------+---------------------+--------+
| Id      | Database | Table | Columns | Type   | Schedule | Properties                        | Status  | LastWorkTime        | Reason |
+---------+----------+-------+---------+--------+----------+-----------------------------------+---------+---------------------+--------+
| 2558263 | ALL      | ALL   | ALL     | SAMPLE | SCHEDULE | {}                                | PENDING | None                |        |
| 7109666 | ALL      | ALL   | ALL     | SAMPLE | SCHEDULE | {statistic_exclude_pattern=td.a1} | FAILED  | 2023-02-20 10:45:37 |        |
+---------+----------+-------+---------+--------+----------+-----------------------------------+---------+---------------------+--------+
2 rows in set
Time: 0.007s
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 13 (53.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|1|7|14.29%|[167, 170, 171, 172, 173, 174]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|6|6|100.00%|[]|

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsCollectJobFactory.java [hunk 1]
```diff
@@ -107,6 +109,16 @@ private static void createJob(List<StatisticsCollectJob> allTableJobMap, Analyze
             return;
         }
 
+        // check job exclude db.table
+        String regex = job.getProperties().getOrDefault(StatsConstants.STATISTIC_EXCLUDE_PATTERN, null);
+        if (StringUtils.isNotBlank(regex)) {
+            Pattern checkRegex = Pattern.compile(regex);
+            String name = db.getFullName() + "." + table.getName();
+            if (checkRegex.matcher(name).find()) {
+                return;
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsCollectJobFactory.java:118]: Is case insensitive better or not?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsCollectJobFactory.java:118]: I think case sensitive is better？more strick

**Review by packy92 — DISMISSED** [review]

**Review by liuyehcf — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by evelynzhaojie — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 13 (53.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|1|7|14.29%|[167, 170, 171, 172, 173, 174]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|6|6|100.00%|[]|

**Pull request merged**